src/main/java/itm/oss/splitter/Reports.java

```diff
@@ -1,12 +1,28 @@
 package itm.oss.splitter;
 
+import java.math.BigDecimal;
 import java.util.ArrayList;
+import java.util.LinkedHashMap;
 
 public class Reports {
 
   public SimpleMap totalsByCategory(ArrayList<Expense> xs) {
-    // TODO (Issue 6): sum amounts per category
-    throw new UnsupportedOperationException("totalsByCategory() not implemented yet");
+    LinkedHashMap<String, BigDecimal> totals = new LinkedHashMap<>();
+
+    for (int i = 0; i < xs.size(); i++) {
+      Expense e = xs.get(i);
+      String category = e.getCategory();
+      if (category == null)
+        category = "";
```

Comment:
Regarding the consistency of the code, it would be better to enclose the if-condition code in brackets for enhancement of consistency.
```suggestion
      if (category == null) {
        category = "";
      }
```

Reply:
Your review seems reasonable. However, the suggestion by @Poppaluu resolves your comment, so I will apply it naturally by taking his review.

Thank you for your comment!!